keel-schema-generator/src/main/kotlin/com/netflix/spinnaker/keel/schema/ResourceKindSchemaCustomizer.kt

```diff
@@ -0,0 +1,17 @@
+package com.netflix.spinnaker.keel.schema
+
+import com.netflix.spinnaker.keel.api.ResourceKind
+import com.netflix.spinnaker.keel.api.plugins.SupportedKind
+import kotlin.reflect.KClass
+
+class ResourceKindSchemaCustomizer(
+  private val kinds: Iterable<ResourceKind>
+) : SchemaCustomizer {
+
+  override fun supports(type: KClass<*>) = type == ResourceKind::class
+
+  override fun buildSchema() = EnumSchema(
+    description = "The resource kind that corresponds with each spec format.",
+    enum = kinds.map(ResourceKind::toString)
+  )
+}
```

Comment:
This is the only bit of customization necessary. I could even add some kind of annotation for this but this doesn't seem too heinous.